htdocs/core/lib/ajax.lib.php

```diff
@@ -121,11 +121,11 @@ function ajax_autocompleter($selected, $htmlname, $url, $urloption = '', $minLen
 
 					// Activate the autocomplete to execute the GET
     				$("input#search_'.$htmlnamejquery.'").autocomplete({
-    					source: function( request, response ) {
-    						$.get("'.$url.($urloption ? '?'.$urloption : '').'", { "'.str_replace('.', '_', $htmlname).'": request.term }, function(data){
-								if (data != null)
-								{
-									response($.map( data, function(item) {
+						source: function(request, response) {
+							var maxResults = '.(getDolGlobalString('AJAX_LIMIT_AUTOCOMPLETE_RESULTS') ? getDolGlobalString('AJAX_LIMIT_AUTOCOMPLETE_RESULTS') : '10').'; // Dynamically set maxResults
```

Comment:
Setting a limit without user consent lead to user experience trouble: They don't understand why they can't see some results and generates support ticket.
So default value must be 0. And if value is 0, we do no execute data.slice.